### PR TITLE
Fix redirect to language without URL prefix

### LINKED ISSUE #6328: Language redirection with or without URL prefix does not work, no index alias assigned
### Affected version(s)

4.13.29

### Description

In the installation there are two page trees one for NL without URL prefix and one with EN with URL prefix EN. In the page trees there is no page with the alias index.

Language forwarding is disabled for NL, language fallback is enabled for EN.

No matter with which language I call the domain.tld, I always land on the first page of the NL page tree. 
Only when the URL prefix is set at NL, a correct redirection takes place. The language redirection option has no effect.

Legacy routing is disabled

2023-08-23_12h33_25
2023-08-23_12h33_49

### PR DESCRIPTION
**Pull request by aschempp**
https://github.com/contao/contao/issues/6328#issuecomment-1695256820

First commit is the reproduction test, second commit fixes the issue.

### CODE REVIEW

**Comment by ausi** [comment]
But shouldn’t we also ensure that it works correctly without `disableLanguageRedirect`, so that browsers with language nl get redirected to `/home` and english browsers to `/en/home`?

**Comment by aschempp** [comment]
By _ensure_ you mean to write a test? Are you sure that's not already there? Because we have a _lot_ of routing tests? 😂

**Comment by ausi** [comment]
> By _ensure_ you mean to write a test?

I thought of manually checking if that is the case, but in the end with a test, yes.

> Are you sure that's not already there?

No, I’m not sure. But according to the issue description in #6328 it did not work that way, so I’m pretty certian that this is not tested yet.

**Comment by aschempp** [comment]
I've added a lot more tests in 8de7a0f99435285a77a46fb2952c6dc4ca49596a. The comments above each `yield` in the dataProvider are relevant.

The behaviour is consistent even though it might sometimes be unexpected. For example, in the given case (en prefix = 'en', nl prefix = empty), visitors will **always** see the dutch home page (regardless of browser locale), unless `disableLanguageRedirect` is enabled on the dutch root. If `disableLanguageRedirect` is enabled, visitors will **never** see the dutch home page (regardless of browser locale).

File: core-bundle/src/Routing/RouteProvider.php [hunk 1]
```diff
@@ -157,6 +157,7 @@ private function addRoutesForRootPages(array $pages, array &$routes): void
     private function createCollectionForRoutes(array $routes, array $languages): RouteCollection
     {
         $this->sortRoutes($routes, $languages);
+        dump(array_keys($routes));
```
Comment (Toflar) [inline core-bundle/src/Routing/RouteProvider.php]:
```suggestion
```

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Pull request merged**